pkg/api/v1/clients/kube/controller/controller.go

```diff
@@ -17,132 +17,112 @@ import (
 	"k8s.io/client-go/util/workqueue"
 )
 
+// This custom Kubernetes controller is used to provide a shared caching mechanism for the solo-kit resource clients.
 type Controller struct {
 	name string
 
-	kubeclientset kubernetes.Interface
+	informers []cache.SharedInformer
 
-	syncFuncs []cache.InformerSynced
-
-	// workqueue is a rate limited work queue. This is used to queue work to be
+	// WorkQueue is a rate limited work queue. This is used to queue work to be
 	// processed instead of performing it as soon as a change happens. This
 	// means we can ensure we only process a fixed amount of resources at a
 	// time, and makes it easy to ensure we are never processing the same item
 	// simultaneously in two different workers.
-	workqueue workqueue.RateLimitingInterface
+	workQueue workqueue.RateLimitingInterface
 
-	// recorder is an event recorder for recording Event resources to the
-	// Kubernetes API.
+	// recorder is an event recorder for recording Event resources to the Kubernetes API
 	recorder record.EventRecorder
 
 	// handler to call
 	handler cache.ResourceEventHandler
 }
 
-// NewController returns a new controller
+// Returns a new kubernetes controller without starting it. The given event handler is registered with each of the
+// given informers. The controller also collects the HasSynced functions of each of the informers.
 func NewController(
 	controllerName string,
-	kubeclient kubernetes.Interface,
+	kubeClient kubernetes.Interface,
 	handler cache.ResourceEventHandler,
 	informers ...cache.SharedInformer) *Controller {
 
+	// TODO: how will this work with unauthorized users?
 	eventBroadcaster := record.NewBroadcaster()
```

Comment:
@ilackarms @yuval-k It looks like this `recorder` is currently not used. Right now the controller works and I did not investigate further. I can see how this would fail in case `kubeclient` does not have permissions on `Event`s. What do you think?

Reply:
Ok, will do